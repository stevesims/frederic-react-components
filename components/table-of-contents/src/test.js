import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';

import TableOfContents from '.';

const renderIndex = ({ contents }) =>
  contents.map(({ id, title }) => <div id={`index-${id}`} key={id}>{id}: {title}</div>);

describe('TableOfContents', () => {
  it('renders without crashing', () => {
    const wrapper = mount(
      <TableOfContents.Container>
        <TableOfContents.Section id="id" title="Title">Section text here</TableOfContents.Section>
        <TableOfContents.Index>{ renderIndex }</TableOfContents.Index>
      </TableOfContents.Container>
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('can remove a section', () => {
    const TestContainer = ({ showSection }) => (
      <TableOfContents.Container>
        <TableOfContents.Section id="id" title="Title">Section text here</TableOfContents.Section>
        { showSection &&
        <TableOfContents.Section id="optional" title="Optional">Optional section here</TableOfContents.Section>
      }
        { showSection &&
        <TableOfContents.Section id="optional" title="Optional">Duplicate section here</TableOfContents.Section>
      }
        <TableOfContents.Index>{ renderIndex }</TableOfContents.Index>
      </TableOfContents.Container>
    );
    TestContainer.propTypes = {
      showSection: PropTypes.bool,
    };

    const wrapper = mount(<TestContainer showSection />);

    expect(wrapper.find('#index-optional').length).toBe(1);

    wrapper.setProps({ showSection: false });
    wrapper.update();

    expect(wrapper.find('#index-optional').length).toBe(0);
  });

  it('matches snapshot', () => {
    const wrapper = mount(
      <TableOfContents.Container>
        <TableOfContents.Section id="id" title="Title">Section text here</TableOfContents.Section>
        <TableOfContents.Index>{ renderIndex }</TableOfContents.Index>
      </TableOfContents.Container>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
