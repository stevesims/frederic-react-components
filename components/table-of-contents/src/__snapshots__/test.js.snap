// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOfContents matches snapshot 1`] = `
<Container>
  <Section
    id="id"
    title="Title"
  >
    <Entry
      id="id"
      title="Title"
      toc={
        Object {
          "addContent": [Function],
          "contents": Array [
            Object {
              "id": "id",
              "title": "Title",
            },
          ],
          "removeContent": [Function],
        }
      }
    >
      <div
        id="id"
      >
        Section text here
      </div>
    </Entry>
  </Section>
  <div
    id="index-id"
    key="id"
  >
    id
    : 
    Title
  </div>
</Container>
`;
